portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,26 +33,43 @@
 /** Servlet that returns comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  
-  // Create ArrayList for comments
-  List<String> comments = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Convert ArrayList of comments to JSON using Gson
-    String json = new Gson().toJson(comments);
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String newComment = (String) entity.getProperty("text");
+      long time = (long) entity.getProperty("time");
```

Comment:
You could use [`Entity.getString(name)`](https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/datastore/BaseEntity.html#getString-java.lang.String-) and [`Entity.getLong(name)`](https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/datastore/BaseEntity.html#getLong-java.lang.String-) instead:
```suggestion
      String newComment = entity.getString("text");
      long time = entity.getLong("time");
```

There's also [`Entity.getTimestamp(name)`](https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/datastore/BaseEntity.html#getTimestamp-java.lang.String-) but that returns a  [`com.google.cloud.Timestamp`](https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/Timestamp.html), which you could use in place of or together with `java.util.Date`.